### DISCUSSION TITLE
time_ns() vs time() on ESP32-S2

### DISCUSSION THREAD

**Discussion by fcolmant** [micropython/micropython#10791] [ESP32]
Hello,

I'm developing on the ESP32-S2 and encounter a strange behaviour:
The code:
```
import time
while True :
    s_time = time.time()
    ns_time = int(time.time_ns()/1e09)
    print('time():')
    print(s_time)
    print('time_ns():')
    print(ns_time)
    print('delta:')
    print(s_time - ns_time)
    time.sleep(1)
```

When I run this code after a restart, time.time_ns() is frozen. EG:
`>>> %Run -c $EDITOR_CONTENT
time():
730130955
time_ns():
730130944
delta:
11
time():
730130956
time_ns():
730130944
delta:
12`

When I run it after a brownout, time.time_ns() is correctly implemented:
`time():
2
time_ns():
2
delta:
0
time():
3
time_ns():
3
delta:
0
time():
4
time_ns():
4
delta:
0`

Any idea?

**Comment by robert-hh** [comment]
That's strange, since both calls used the same call to gettimeofday(). Only time_ns() adds the µs value as well.. Try running it without Thonny.

**Reply by fcolmant** [reply]
Good idea, which I tried with REPL over serial. No success however.

**Reply by robert-hh** [reply]
No problem here with a ESP32S2, ESP32, ESP32C3. The above code, pasted with Ctrl-E, Ctrl-D shows a proper time series after boot. E.g.:
time():
10
time_ns():
10
delta:
0
time():
11
time_ns():
11
delta:
0
time():
12
time_ns():
12
delta:
0
